napari/utils/events/custom_types.py

```diff
@@ -35,7 +35,16 @@ def validate_type(cls, val):
         else:
             shape = ()
 
-        result = np.array(val, dtype=dtype, copy=False, ndmin=len(shape))
+        # In numpy 1, copy=False means that copy is only made if needed.
+        # In numpy 2, copy=False errors if a copy is needed, and instead
+        # the equivalent is copy=None.
+        # We would normally use np.asarray instead, but here we need the
+        # behavior associated with ndmin.
+        # https://numpy.org/devdocs/numpy_2_0_migration_guide.html#adapting-to-changes-in-the-copy-keyword
+        copy = None
+        if np.lib.NumpyVersion(np.__version__) < '2.0.0b1':
```

Comment:
There's probably a better comparison than this, but I'm sleepy and this is the best I've got right now.